subprojects/test-runner/device-provider/impl/src/main/kotlin/com/avito/android/runner/devices/internal/kubernetes/KubernetesReservationReleaser.kt

```diff
@@ -38,6 +38,19 @@ internal class KubernetesReservationReleaser(
         }
     }
 
+    suspend fun releaseDeployment(
+        deploymentName: String
+    ) {
+        with(CoroutineScope(coroutineContext + CoroutineName("delete-deployment-$deploymentName"))) {
+            launch {
+                kubernetesApi.getPods(deploymentName)
+                    .onSuccess { pods -> releasePods(deploymentName, pods) }
+                    .onFailure { error -> logger.warn("Can't get pods when release", error) }
+                kubernetesApi.deleteDeployment(deploymentName)
```

Comment:
In current behaviour is problem. When we create a deployment we call `deploymentPodsListener.start`. This start coroutine that poll deployment for new pods. This couroutine should be stopped gracefully after deployment delition